third-party/rsutils/src/shorten-json-string.cpp

```diff
@@ -0,0 +1,128 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include <rsutils/string/shorten-json-string.h>
+
+
+namespace rsutils {
+namespace string {
+
+
+// Given an outside block, e.g.:
+//        012345678901234567890123456789012345678901234567890123456789012345678901234567890
+//        0         1         2         3         4         5         6         7         8
+//        {"one":1,"two":[1,2,3],"three":{"longassblock":{"insideblock":89012}},"four":4}
+// Find the first inside block, including the enclosing delimiters (parenthesis or square brackets):
+//                       ^______^        ^_____________________________________^
+//
+static slice find_inside_block( slice const & outside )
+{
+    // find an opening
+    slice::const_iterator it = outside.begin() + 1;  // opening {, separating comma, etc.
+    bool in_quote = false;
+    while( it < outside.end() && ( in_quote || *it != '[' && *it != '{' ) )
```

Comment:
It is better to move `*it != '[' && *it != '{'` to a separate function `is_opening`, the meaning will be clearer.
For the very least surround with parenthesis, don't count on operator `&&` precedence over `||`.

Reply:
I don't like parens that aren't needed generally... I'll try to add `is_opening`

Reply:
Fixed